### ISSUE TITLE
Module quick reference page

### ISSUE THREAD

**Issue by jfy133** [nf-core/nf-co.re#834]
**Is your feature request related to a problem? Please describe.**
When writing a pipeline, it's workflow-breaking to have to hunt through loads of directories in your IDE/explorer to get the `meta.yaml` file of a module, when you have many modules installed.

**Describe the solution you'd like**
A page on the website to rendering of the (currently otherwise unused?) `meta.yml` of all modules, allowing quick search to find the information you need when writing the pipeline (as often I have a browser open next to me IDE for googling).

**Describe alternatives you've considered**
Not be so lazy, or a `nf-core/tools` command to display it for you.

**Additional context**
@ewels volunteered pending reminders: https://nfcore.slack.com/archives/CE7DN1U7M/p1628502439008800

**Comment by mashehu** [comment]
I guess we could make it similar to https://nf-co.re/pipelines, with each module being represented by one card, with all the info on it and  tool-/sidebar to search and filter.

**Comment by jfy133** [comment]
That would be exactly what I'm looking for :ok_hand:

**Comment by ewels** [comment]
I wondered that at first, but I think that there will be too many. A list view (as we also have on the pipeline page) would make it easier to scan the long list. Also have a quick-search bar that highlights partial matches for fast visual grepping. And can think about what info we want from the meta file to display in the list.

Once this page is done we should also make a dynamic page for each module showing all the info that we have. Stuff in the meta file, maybe even show which subworkflows (and pipelines??) the module is used in - that kind of stuff would be ace. Also the history with git commit messages and hashes with quick-copy buttons.

**Comment by mashehu** [comment]
The positive thing about a grid view is that you can show more information closer together and easier separable from others. I was anyway imagining the main way of interacting with it through the filters/search (which would need to be quite prominent because of that) and not by scrolling through. But maybe my head just went too fast towards the interfaces of iftt or siri shortcuts. 
For people being interested in following parts of a pipeline for a new one, we could think about adding the pipeline names as filter categories, so you could click on cageseq and see all the modules in it (best case even in the correct order 🤓).

Because modules are now some kind of their own pipeline, I was also thinking of having a page for each. But I don't think it needs to be as large as the pipeline pages. I am thinking more like the single page npm sites (which I think the pipeline pages are designed after, but then they got more and more tabs 😄). I see at the end we will get another leaderboard, but this time for modules 😉.

**Comment by jfy133** [comment]
Honestly I think you should replicate exactly what you have for the pipeline page.

The current pipeline page which is one (mobile) to two column grid is fine enough for me to identify exactly what I want after using the search bar that's already there on the top. You already have a one line description and meta tags in the `meta.yml` in all the modules already.

Ultimately you would have the same UX on both pages which means users will be familiar how both works. I don't think there is any need to reinvent the wheel there.

I guess the _only_ thing I would _maybe_ think could change is having the input/output descriptions for each module as a collapseable component of each card (so you don't have to open an entirely new tab), as in most cases the `meta.yml` info is not that extensive. But that's only a cherry-on-top- sort of thing which isn't necessary.

**Comment by ewels** [comment]
Yeah fair enough! If we mimic the existing pipeline page then we have the table view as well anyway  so can have best of both worlds :) And can add the visual search match highlighting to both pages I guess.

I think that we should allow the module pages to be flexible enough to grow - there will certainly be more stuff we want to add as time goes on.. 😅

**Comment by maxulysse** [comment]
I'd love to have that for the header fields in the samplesheet too, and for the main params as well

**Comment by ewels** [comment]
@maxulysse have what sorry?

**Comment by maxulysse** [comment]
A page on the website that have all params used in all pipelines, so that I'm sure that the names are consistent.
And another page with all the headers in the csv samplesheets, so that I can choose a good name too

**Comment by ewels** [comment]
Interesting idea! 🤔 I think that deserves its own issue though..
